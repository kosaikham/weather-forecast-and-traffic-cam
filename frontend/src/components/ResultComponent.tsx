import React from "react";
import { Image, Alert } from "antd";
import { APIResponse } from "../types";

interface ResultComponentProps {
  selectedOption: APIResponse;
  visible: boolean;
}

export const ResultComponent: React.FC<ResultComponentProps> = ({
  selectedOption,
  visible,
}: ResultComponentProps) => {
  return visible ? (
    <div className="content-body">
      <div>
        <Alert message={selectedOption.weather} type="info" showIcon />
      </div>
      <div className="image">
        <Image width="100%" preview={false} src={selectedOption.image} />
      </div>
    </div>
  ) : null;
};
