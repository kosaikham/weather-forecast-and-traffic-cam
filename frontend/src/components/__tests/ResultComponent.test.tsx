import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ResultComponent } from '../ResultComponent';

describe('<ResultComponent />', () => {
  it('has the correct text', () => {
    const selectedOption = {
      name: 'Marine Parade',
      weather: 'Cloudy',
      location: { latitude: 1.297512569, longitude: 103.8983019 },
      image: 'https://images.data.gov.sg/api/traffic-images/2023/07/a11f21c0-382d-4293-8008-56cd47f485f6.jpg',
      id: 36
    }
    render(<ResultComponent selectedOption={selectedOption} visible/>);
    expect(screen.getByText('Cloudy')).toBeTruthy();
  });
});

describe('<ResultComponent /> snapshot', () => {
  it('renders correctly', () => {
    const selectedOption = {
      name: 'Marine Parade',
      weather: 'Cloudy',
      location: { latitude: 1.297512569, longitude: 103.8983019 },
      image: 'https://images.data.gov.sg/api/traffic-images/2023/07/a11f21c0-382d-4293-8008-56cd47f485f6.jpg',
      id: 36
    }
    const tree = renderer.create(<ResultComponent selectedOption={selectedOption} visible />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});