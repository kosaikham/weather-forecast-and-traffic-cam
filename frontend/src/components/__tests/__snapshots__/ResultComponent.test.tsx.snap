// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultComponent /> snapshot renders correctly 1`] = `
<div
  className="content-body"
>
  <div>
    <div
      className="ant-alert ant-alert-info css-dev-only-do-not-override-1m62vyb"
      data-show={true}
      role="alert"
      style={Object {}}
    >
      <span
        aria-label="info-circle"
        className="anticon anticon-info-circle ant-alert-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="info-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
          />
        </svg>
      </span>
      <div
        className="ant-alert-content"
      >
        <div
          className="ant-alert-message"
        >
          Cloudy
        </div>
      </div>
    </div>
  </div>
  <div
    className="image"
  >
    <div
      className="ant-image css-dev-only-do-not-override-1m62vyb"
      style={
        Object {
          "height": undefined,
          "width": "100%",
        }
      }
    >
      <img
        className="ant-image-img css-dev-only-do-not-override-1m62vyb"
        onLoad={[Function]}
        src="https://images.data.gov.sg/api/traffic-images/2023/07/a11f21c0-382d-4293-8008-56cd47f485f6.jpg"
        style={
          Object {
            "height": undefined,
          }
        }
        width="100%"
      />
    </div>
  </div>
</div>
`;
